From: Rene Kita <mail@rkta.de>
Date: Tue Sep 17 12:22:09 CEST 2024
Subject: Get rid of STR_LENGTH_CHECK
Patch: patches/Get-rid-of-STR_LENGTH_CHECK.patch

Code even has a comment that it is obsolete. Initial revision in CVS
from 2001 already has the comment.
---
 Str.c |   28 ----------------------------
 1 file changed, 28 deletions(-)

--- a/Str.c
+++ b/Str.c
@@ -26,13 +26,6 @@
 
 #define INITIAL_STR_SIZE 32
 
-#ifdef STR_DEBUG
-/* This is obsolete, because "Str" can handle a '\0' character now. */
-#define STR_LENGTH_CHECK(x) if (((x)->ptr==0&&(x)->length!=0)||(strlen((x)->ptr)!=(x)->length))abort();
-#else				/* not STR_DEBUG */
-#define STR_LENGTH_CHECK(x)
-#endif				/* not STR_DEBUG */
-
 Str
 Strnew(void)
 {
@@ -139,7 +132,6 @@ Str
 Strdup(Str s)
 {
     Str n = Strnew_size(s->length);
-    STR_LENGTH_CHECK(s);
     Strcopy(n, s);
     return n;
 }
@@ -161,8 +153,6 @@ Strfree(Str x)
 void
 Strcopy(Str x, Str y)
 {
-    STR_LENGTH_CHECK(x);
-    STR_LENGTH_CHECK(y);
     if (x->area_size < y->length + 1) {
 	x->ptr = GC_REALLOC(x->ptr, y->length + 1);
 	if (x->ptr == NULL)
@@ -178,7 +168,6 @@ Strcopy_charp(Str x, const char *y)
 {
     int len;
 
-    STR_LENGTH_CHECK(x);
     if (y == NULL) {
 	x->length = 0;
 	x->ptr[0] = '\0';
@@ -203,7 +192,6 @@ Strcopy_charp_n(Str x, const char *y, in
 {
     int len = n;
 
-    STR_LENGTH_CHECK(x);
     if (y == NULL) {
 	x->length = 0;
 	x->ptr[0] = '\0';
@@ -227,7 +215,6 @@ Strcat_charp_n(Str x, const char *y, int
 {
     int newlen;
 
-    STR_LENGTH_CHECK(x);
     if (y == NULL || n == 0)
 	return;
     if (n < 0)
@@ -256,7 +243,6 @@ Strcat_charp_n(Str x, const char *y, int
 void
 Strcat(Str x, Str y)
 {
-    STR_LENGTH_CHECK(y);
     Strcat_charp_n(x, y->ptr, y->length);
 }
 
@@ -312,7 +298,6 @@ Strsubstr(Str s, int beg, int len)
     Str new_s;
     int i;
 
-    STR_LENGTH_CHECK(s);
     new_s = Strnew();
     if (beg >= s->length)
 	return new_s;
@@ -325,7 +310,6 @@ void
 Strlower(Str s)
 {
     int i;
-    STR_LENGTH_CHECK(s);
     for (i = 0; i < s->length; i++)
 	s->ptr[i] = TOLOWER(s->ptr[i]);
 }
@@ -334,7 +318,6 @@ void
 Strupper(Str s)
 {
     int i;
-    STR_LENGTH_CHECK(s);
     for (i = 0; i < s->length; i++)
 	s->ptr[i] = TOUPPER(s->ptr[i]);
 }
@@ -342,7 +325,6 @@ Strupper(Str s)
 void
 Strchop(Str s)
 {
-    STR_LENGTH_CHECK(s);
     while (s->length > 0 &&
 	   (s->ptr[s->length - 1] == '\n' || s->ptr[s->length - 1] == '\r')) {
 	s->length--;
@@ -354,7 +336,6 @@ void
 Strinsert_char(Str s, int pos, char c)
 {
     int i;
-    STR_LENGTH_CHECK(s);
     if (pos < 0 || s->length < pos)
 	return;
     if (s->length + 2 > s->area_size)
@@ -370,7 +351,6 @@ Strinsert_char(Str s, int pos, char c)
 void
 Strinsert_charp(Str s, int pos, const char *p)
 {
-    STR_LENGTH_CHECK(s);
     while (*p)
 	Strinsert_char(s, pos++, *(p++));
 }
@@ -379,7 +359,6 @@ void
 Strdelete(Str s, int pos, int n)
 {
     int i;
-    STR_LENGTH_CHECK(s);
     if (pos < 0 || s->length < pos)
 	return;
     if (n < 0)
@@ -398,7 +377,6 @@ Strdelete(Str s, int pos, int n)
 void
 Strtruncate(Str s, int pos)
 {
-    STR_LENGTH_CHECK(s);
     if (pos < 0 || s->length < pos)
 	return;
     s->ptr[pos] = '\0';
@@ -408,7 +386,6 @@ Strtruncate(Str s, int pos)
 void
 Strshrink(Str s, int n)
 {
-    STR_LENGTH_CHECK(s);
     if (n >= s->length) {
 	s->length = 0;
 	s->ptr[0] = '\0';
@@ -424,7 +401,6 @@ Strremovefirstspaces(Str s)
 {
     int i;
 
-    STR_LENGTH_CHECK(s);
     for (i = 0; i < s->length && IS_SPACE(s->ptr[i]); i++) ;
     if (i == 0)
 	return;
@@ -436,7 +412,6 @@ Strremovetrailingspaces(Str s)
 {
     int i;
 
-    STR_LENGTH_CHECK(s);
     for (i = s->length - 1; i >= 0 && IS_SPACE(s->ptr[i]); i--) ;
     s->length = i + 1;
     s->ptr[i + 1] = '\0';
@@ -448,7 +423,6 @@ Stralign_left(Str s, int width)
     Str n;
     int i;
 
-    STR_LENGTH_CHECK(s);
     if (s->length >= width)
 	return Strdup(s);
     n = Strnew_size(width);
@@ -464,7 +438,6 @@ Stralign_right(Str s, int width)
     Str n;
     int i;
 
-    STR_LENGTH_CHECK(s);
     if (s->length >= width)
 	return Strdup(s);
     n = Strnew_size(width);
@@ -480,7 +453,6 @@ Stralign_center(Str s, int width)
     Str n;
     int i, w;
 
-    STR_LENGTH_CHECK(s);
     if (s->length >= width)
 	return Strdup(s);
     n = Strnew_size(width);
